core/input/input.cpp

```diff
@@ -1505,6 +1510,29 @@ Input::Input() {
 			parse_mapping(entries[i]);
 		}
 	}
+
+	String env_ignore_devices = OS::get_singleton()->get_environment("SDL_GAMECONTROLLER_IGNORE_DEVICES");
+	if (!env_ignore_devices.is_empty()) {
+		Vector<String> entries = env_ignore_devices.split(",");
+		for (int i = 0; i < entries.size(); i++) {
+			Vector<String> vid_pid = entries[i].split("/");
+
+			if (vid_pid.size() < 2) {
+				continue;
+			}
+
+			print_verbose(vformat("Device Ignored -- Vendor: %s Product: %s", vid_pid[0], vid_pid[1]));
+			const uint16_t vid_unswapped = vid_pid[0].hex_to_int();
+			const uint16_t pid_unswapped = vid_pid[1].hex_to_int();
+
+			const uint16_t vid = BSWAP16(vid_unswapped);
+			if (ignored_vid_pid_devices.has(vid)) {
```

Comment:
You can use the first branch of the `if` directly, since `HashMap::operator[]` will add the entry if not already present.